addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ModernAppDetectionScanRuleUnitTest.java

```diff
@@ -37,6 +37,19 @@ protected ModernAppDetectionScanRule createScanner() {
         return new ModernAppDetectionScanRule();
     }
 
+    @Test
+    void shouldNotRaiseAlertWithNonHtmlContent() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        msg.setResponseHeader("HTTP/1.1 200\r\n" + "Content-Type: application/json\r\n");
+        msg.setResponseBody("{\"content\": \"anything\"}");
```

Comment:
For my two cents I'd suggest making this something that normally would be Alerted on,

Reply:
@kingthorin Do you mean that ModernAppDetectionScanRule should raise an alert on any JSON content?

I can do that, maybe in another PR, because the goal here is to fix the existing code that should not raise an alert for non-HTML content...

What content-type would you suggest for this new test? In my practical case zaproxy/zaproxy/issues/7617 it was JPEG. Maybe I can just set the responseHeader to "image/jpeg" and no responseBody?

Best regards

Reply:
No no, I was just suggesting to make the Body content something that the rule would normally catch. (So that the significance of the change/test is more illustrative.) The content type can be `application/foo` or just about anything if you’re worried about consistency.

Reply:
I dunno, @thc202 am I over thinking this?

Reply:
You aren't, as is the test passes whether the fix is in place or not.

Reply:
My bad, I did misunderstood your comment and indeed, my test was passing before the fix.

I changed the content-type to application/foo and the content to `"content that would normally raise <a> an alert"` because my initial bug report is that `<a>` in a JPEG content raises the alert.

Reply:
Seems fine to me.
![image](https://user-images.githubusercontent.com/7570458/204137341-e1a98dec-9ae2-4155-b647-28800887bdca.png)